src/components/BotBuilder/BotBuilder.js

```diff
@@ -202,11 +212,29 @@ class BotBuilder extends React.Component {
         this.setState({
           openSnackbar: true,
           msgSnackbar: 'Unable to delete your draft. Please try again.',
+          deleteAlert: null,
         });
       }.bind(this),
     });
   };
 
+  openDeleteAlert = (type, params) => {
+    this.setState({ deleteAlert: { type, params } });
+  };
+
+  cancelDeleteAlert = () => {
+    this.setState({ deleteAlert: null });
+  };
+
+  closeDeleteAlert = event => {
+    const { type, params } = this.state.deleteAlert;
+    if (type === 'bot') {
+      this.deleteBot(...params);
+    } else if (type === 'draft') {
+      this.deleteDrafts(...params);
+    }
```

Comment:
After deleting the bot/draft, also make sure to close the dialog, by calling the `closeDeleteAlert` (after renaming) function.